### PR TITLE
Make colony home page mobile responsive

### LINKED ISSUE #3475: Create mobile responsive Colony Home page
### Story

When using Colony on my mobile, I would like to use a mobile-optimised version of the dapp.

### Description 

This issue focuses on the first screen of the [Figma design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide?node-id=6641%3A9676) (see below). Namely, adding mobile responsiveness to the Colony's home page. 

### Design 

**Screen 1: Colony Home Page**

colony-home-des

### This issue is complete when...

- [x] Visiting the Colony home page on screens less than or equal to `768px` looks like the design above.
- [x] All popovers and menus accessible from the home page should fit to mobile
- [x] Logging out and logging in again should work on mobile
- [x] Creating new actions should work on mobile

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR is the feature for a mobile responsive Colony Home page. It accords with Screen 1 of the [Figma design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide?node-id=6641%3A9676). 

Note: it does not handle the wiring of the hamburger menu or the rewiring of the user avatar (colony summary) popover menu (i.e. screens 3 and 7). It also does not deal with any of the other screens. These will all be included in further PRs. It's purely focused on the layout of the Colony home page. 

**New stuff** ✨

- [ ] #3486 
- [ ] #3495
- [ ] #3499

## Screenshots 

**_Signed In_**
colony-home

**_Signed Out_**
colony-home-signout

## TODO

Sections To Complete:

- [x] Colony Avatars / User Avatar / Hamburger Menu
- [x] Colony Info / Join Colony
  - [x] Align colony info popover 
- [x] Funds info / manage funds
  - [x] Align coin info popover 
- [x] Actions menu
  - [x] Align create new team menu    
  - [x] Align new action menu
- [x] Actions summary
  - [x] Add load more button for actions
- [x] Connect wallet (signed out)
  - [x] Align button 
  - [x] Adjust menu

Resolves: #3475 
Resolves: #3486 
Resolves: #3495
Resolves: #3499

### CODE REVIEW

**Comment by dit7ya** [comment]
This is huge, whoa! Are we sure there is no other simpler way to achieve this without external libraries? (It might very well be the case but I am just double-checking).

**Comment by willm30** [comment]
> This is huge, whoa! Are we sure there is no other simpler way to achieve this without external libraries? (It might very well be the case but I am just double-checking).

I went for [react-responsive](https://github.com/yocontra/react-responsive) mainly for peace of mind. We could write our own hook to detect screen size I guess but using the library was quicker and probably more reliable short-term (better tested).

I guess we generally try to avoid external libraries where possible?

File: src/modules/pages/components/RouteLayouts/Default/Default.tsx [hunk 1]
```diff
@@ -45,10 +51,24 @@ const Default = ({
...
+                  <SubscribedColoniesList />
+                </div>
+              )}
             </div>
+          ) : (
+            hasSubscribedColonies && (
+              <div className={styles.coloniesList}>
+                <SubscribedColoniesList />
+              </div>
+            )
```
Comment (rdig) [inline src/modules/pages/components/RouteLayouts/Default/Default.tsx:71]: Please try to reduce code repetition as much as possible:
```suggestion
const SubscribedColonies = () => hasSubscribedColonies ? (
    <div className={styles.coloniesList}>
      <SubscribedColoniesList />
    </div>
  ) : null;

{isMobile ? (
  // Render the UserNavigation and SubscribedColoniesList in shared parent on mobile
  <div className={styles.head}>
    <div className={navStyles.nav}>
      <UserNavigation />
    </div>
    {SubscribedColonies()}
  </div>
) : SubscribedColonies()
```

**Review by rdig — CHANGES_REQUESTED** [review]
This is mostly good and kudos to you for the amount of work you've put in. :100: 

Other than what I've picked up in the review _(see comments)_, there's also some other things need discussing:

1. We've changed the design specs for the _Joined Colonies_ list, so your implementation is no longer up to spec. You can tackle it here, or in a separate PR, up to you.

2. There's a discussion on how tread intermediate view port sizes, and if we should cater to them _(ie: tablets / smaller laptops)_, since the view is good for vertical view phones, >1200 desktop sizes, but broken for everything in between.

So I'd suggest the 3 of us _(me, you, Arren)_ get together in a call an hash this out.

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx [hunk 2]
```diff
@@ -81,13 +87,14 @@ const ColonySubscription = ({
           onClick={() => subscribe()}
           appearance={{ theme: 'blue', size: 'small' }}
           data-test="joinColonyButton"
+          className={styles.joinButton}
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx:91]: This I saw after I submitted the PR and addressed it in the other PR I submitted. Am I ok to leave it in the other one?

**Comment by willm30** [comment]
> 1. We've changed the design specs for the _Joined Colonies_ list, so your implementation is no longer up to spec. You can tackle it here, or in a separate PR, up to you.

I will address it in a separate PR.

**Comment by willm30** [comment]
@rdig I have implemented the changes requested, ready for a re-review and updated the PR body (inc updated screenshots). Will split out screen 2 (joined colonies list) into separate PR.

**Review by arrenv — APPROVED** [review]
It is looking good to me, just a couple of comments.

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx [hunk 3]
```diff
@@ -87,7 +87,7 @@ const ColonySubscription = ({
       )}
       {!isSubscribed && !username && (
         <Link
-          className={styles.createUserRedirect}
+          className={`${styles.createUserRedirect}`}
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx]: Good spot

**Review by rdig — APPROVED** [review]
All good now. Nice work Will! :tada:

**Comment by rdig** [comment]
Also @willm30 as per Github's specs liked issues should be one per line.

So this:
Screenshot from 2022-06-21 23-05-44

Should be written as:
```
Resolves #3475
Resolves #3486
Resolves #3495
Resolves #3499
```

See more: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Pull request merged**